py-polars/tests/unit/datatypes/test_array.py

```diff
@@ -291,6 +291,10 @@ def test_recursive_array_dtype() -> None:
     s = pl.Series(np.arange(6).reshape((2, 3)), dtype=dtype)
     assert s.dtype == dtype
     assert s.len() == 2
+    dtype = pl.Array(pl.List(pl.Array(pl.Int8, (2, 2))), 2)
```

Comment:
Does this work with even more nesting? Can you add a tripple and quadruple nested array?